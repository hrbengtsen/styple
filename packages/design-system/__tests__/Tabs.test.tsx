import * as React from "react";
import { render, RenderResult } from "@testing-library/react";
import { Tabs, TabsList, TabsTrigger, TabsContent } from "..";

describe("Test Tabs", () => {
  it("should render as expected", () => {
    const view: RenderResult = render(
      <Tabs defaultValue="tab1">
        <TabsList>
          <TabsTrigger value="tab1">Account</TabsTrigger>
          <TabsTrigger value="tab2">Settings</TabsTrigger>
        </TabsList>
        <TabsContent value="tab1">Account tab content is here</TabsContent>
        <TabsContent value="tab2">Settings tab content is here</TabsContent>
      </Tabs>
    );

    expect(view).toMatchSnapshot();
  });
});
