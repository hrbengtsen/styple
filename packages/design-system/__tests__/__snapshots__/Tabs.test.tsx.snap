// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tabs should render as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="c-cFaYiI"
        data-orientation="horizontal"
        dir="ltr"
      >
        <div
          aria-orientation="horizontal"
          class="c-ihnDDS"
          data-orientation="horizontal"
          role="tablist"
          style="outline: none;"
          tabindex="0"
        >
          <button
            aria-controls="radix-:r0:-content-tab1"
            aria-selected="true"
            class="c-iIbSyB"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="active"
            id="radix-:r0:-trigger-tab1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Account
          </button>
          <button
            aria-controls="radix-:r0:-content-tab2"
            aria-selected="false"
            class="c-iIbSyB"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="inactive"
            id="radix-:r0:-trigger-tab2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Settings
          </button>
        </div>
        <div
          class="c-cWhwNz c-cWhwNz-dbVHzF-size-fill"
          data-orientation="horizontal"
          role="separator"
        />
        <div
          aria-labelledby="radix-:r0:-trigger-tab1"
          class="c-iCEDLM"
          data-orientation="horizontal"
          data-state="active"
          id="radix-:r0:-content-tab1"
          role="tabpanel"
          style="animation-duration: 0s;"
          tabindex="0"
        >
          Account tab content is here
        </div>
        <div
          aria-labelledby="radix-:r0:-trigger-tab2"
          class="c-iCEDLM"
          data-orientation="horizontal"
          data-state="inactive"
          hidden=""
          id="radix-:r0:-content-tab2"
          role="tabpanel"
          tabindex="0"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="c-cFaYiI"
      data-orientation="horizontal"
      dir="ltr"
    >
      <div
        aria-orientation="horizontal"
        class="c-ihnDDS"
        data-orientation="horizontal"
        role="tablist"
        style="outline: none;"
        tabindex="0"
      >
        <button
          aria-controls="radix-:r0:-content-tab1"
          aria-selected="true"
          class="c-iIbSyB"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-state="active"
          id="radix-:r0:-trigger-tab1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Account
        </button>
        <button
          aria-controls="radix-:r0:-content-tab2"
          aria-selected="false"
          class="c-iIbSyB"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-state="inactive"
          id="radix-:r0:-trigger-tab2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Settings
        </button>
      </div>
      <div
        class="c-cWhwNz c-cWhwNz-dbVHzF-size-fill"
        data-orientation="horizontal"
        role="separator"
      />
      <div
        aria-labelledby="radix-:r0:-trigger-tab1"
        class="c-iCEDLM"
        data-orientation="horizontal"
        data-state="active"
        id="radix-:r0:-content-tab1"
        role="tabpanel"
        style="animation-duration: 0s;"
        tabindex="0"
      >
        Account tab content is here
      </div>
      <div
        aria-labelledby="radix-:r0:-trigger-tab2"
        class="c-iCEDLM"
        data-orientation="horizontal"
        data-state="inactive"
        hidden=""
        id="radix-:r0:-content-tab2"
        role="tabpanel"
        tabindex="0"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
